include/net/net_pkt.h

```diff
@@ -2399,6 +2399,15 @@ static inline int net_pkt_write_le32_new(struct net_pkt *pkt, u32_t data)
 	return net_pkt_write_new(pkt, &data_le32, sizeof(u32_t));
 }
 
+/**
+ * @brief Get the amount of data to be read from current cursor position
+ *
+ * @param pkt Network packet
+ *
+ * @return Amount of data to be read from current pkt cursor
+ */
+size_t net_pkt_data_to_read(struct net_pkt *pkt);
```

Comment:
Can this be named more clearly, e.g. net_pkt_remaining_data_len() or net_pkt_rest_data_len() or net_pkt_tail_data_len()

Reply:
hum, neither nor :)
I am not also super happy of the name I came from. The point here is to get the length of data in front of the current cursor position.

I am open to ideas

Reply:
> The point here is to get the length of data in front of the current cursor position.

That formulation is itself confusing. When you stand *in front of a door*, you stand *before a door*. Here, we talk about data *after* cursor.

I don't see how you don't like any of the 3 options proposed. Call it "net_pkt_data_len_after_cursor()" then, can't beat that in clarity, not so in terseness.

Reply:
still, not great. I am sure we can find something better